src/PLLModelGetLanguagesListDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * Set return type of PLL_Model->get_languages_list().
+ */
+
+declare(strict_types=1);
+
+namespace WPSyntex\Polylang\PHPStan;
+
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use PLL_Language;
+use PLL_Model;
+
+class PLLModelGetLanguagesListDynamicMethodReturnTypeExtension implements DynamicMethodReturnTypeExtension {
+	public function getClass(): string {
+		return PLL_Model::class;
+	}
+
+	public function isMethodSupported( MethodReflection $methodReflection ): bool {
+		return $methodReflection->getName() === 'get_languages_list';
+	}
+
+	public function getTypeFromMethodCall( MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope ): Type {
+		$args = $methodCall->getArgs();
+
+		if ( count( $args ) === 0 ) {
+			// Called without arguments.
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		$argumentType = $scope->getType( $args[0]->value );
+
+		if ( ! $argumentType->isArray()->yes() ) {
+			// Called with an argument that is not an array-like.
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		if ( $argumentType instanceof IntersectionType && $argumentType->isIterable() ) {
```

Comment:
```suggestion
		if ( $argumentType instanceof IntersectionType && $argumentType->isArray()->yes() ) {
```
`IterableType` isn't always an array, also `isFoo()` returns a `TrinaryLogic` which has to be tested via `yes()`, `no()` or `maybe()`.